learning/dashboard/webserver/readme.md

```diff
@@ -0,0 +1,161 @@
+
+#	How to setup and configure Nginx ?
+
+##	Nginx installation on Ubuntu 20.04
+
+A package manager is typically included with every operating system. You have a choice between using apt or apt-get with Ubuntu 20.04. As this is a new server, you first need to update the package manager.
+
+Step 1 :  Upgrade the package management to the most recent version that is compatible:
+You can either use the apt or apt-get package manager to install Nginx.
+'''sudo apt updatesudo apt-get update'''
+Now, you are ready to install Ngnix. When you are logged in with a non-root account with sudo rights, you will need to prefix every task that requires admin privileges with sudo.
+
+Step 2 : Install Ngnix web server using apt:	
+'''sudo apt install nginxsudo apt-get install nginx'''
+
+Although the service starts automatically after installing Nginx, before you can test whether Nginx is working correctly or not, you will need to configure the OS firewall to access Nginx.
+
+## Configure Firewall for Nginx :
+
+The most popular method for creating iptables' ingress and egress rules. Uncomplicated Firewall, or ufw, is a user-friendly wrapper for iptables that is available in Ubuntu 20.04. Although it comes installed with Ubuntu 20.04 but to make sure that it is appropriately installed, install it manually. After installing Nginx, follow the instructions below to set up the firewall
+
+Step 1: Install ufw:
+'''sudo apt install ufw'''
+
+Step 2: Check if ufw is working fine:
+'''sudo ufw status'''
+
+Step 3: Allow traffic over HTTP and recheck the status:
+'''sudo ufw allow ‘Nginx HTTP’ '''
+
+Step 4: Allow traffic over TCP (OpenSSH) too if you want to log into the machine again:
+'''sudo ufw allow ‘OpenSSH’ '''
+
+Step 5: If the status was inactive when you checked earlier, activate it:
+'''sudo ufw enable'''
+
+The choice of whether to implement the changes or not will be presented to you. Be aware that you won't be able to enter into the machine again if you deactivate TCP traffic or add a deny OpenSSH rule to the firewall. Let’s now cover some crucial Nginx commands for administration and management
+
+## Nginx Installation and Management :
+
+Step 1: Check the status of the Nginx server:
+              ''' sudo systemctl status nginx '''
+               
+Step 2: You can also check if Nginx is working by accessing your website using the browser. Visit the following link:
+               http://servermania_server_ip
+               
+Step 3: Use systemctl to start, stop, or restart your Nginx  server:
+'''sudo systemctl stop nginxsudo systemctl start   nginxsudo systemctl restart nginx'''
+my
+Step 4: The Nginx configuration file is found at /etc/nginx/nginx.conf. If you make changes to the configuration, you can either restart the Nginx server or reload the configuration file.
+          '''systemctl reload nginx # reloads Nginx configuration'''
```

Comment:
This doesn't look right. Use single backticks for single line code. Use triple backticks for code blocks.